src/translate-adapter-handlers.ts

```diff
@@ -194,16 +204,95 @@ async function translateIoPackage(): Promise<void> {
 	if (content.common.messages) {
 		console.log("Translate Messages");
 		for (const message of content.common.messages) {
-			console.log(`   Message: ${message.title.en}`);
+			console.log(gray(`   Message: ${message.title.en}`));
 			await translateNotExisting(message.title);
 			await translateNotExisting(message.text);
-			await translateNotExisting(message.linkText);
+			// test first if there is a linkText - since it's not mandatory
+			if (message.linkText) await translateNotExisting(message.linkText);
 		}
 	}
 	await writeJson(ioPackage, content, { spaces: 4, EOL });
 	console.log(`Successfully updated ${path.relative(".", ioPackage)}`);
 }
 
+async function loopJSON(content: object): Promise<object> {
+	const baseContent: object[] = [];
+	// read all existing language files
+	for (let n = 0; n < i18nBases.length; n++) {
+		if (existsSync(i18nBases[n])) {
+			baseContent.push(await readJson(i18nBases[n]));
+		}
+	}
+	// if there is no existing lang file create the default one
+	if (baseContent.length === 0)
+		baseContent[0] = await readJson("./admin/i18n/en/translations.json");
+	let baseContentChanged = false;
+	// iterate over the jsonConfig to find translatable entries
+	for (const [key, value] of Object.entries(content)) {
+		if (key === "i18n" && value === false) {
```

Comment:
We should check this only one root level. based on schema also in another place a i18n coukld be located where it needs to be handled differently 

I would move this check out of this method. and we should also add the option to provide a custom dir. see https://github.com/ioBroker/ioBroker.admin/blob/master/src-rx/src/components/JsonConfigComponent/SCHEMA.md#i18n ... I think it makes sensento add these files then too to i18nBases